arduino/sketch/profiles.go

```diff
@@ -71,18 +90,9 @@ func (p *Profiles) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 
-	var profilesData yaml.MapSlice
-	if err := unmarshal(&profilesData); err != nil {
-		return err
-	}
-
-	for _, profileData := range profilesData {
-		profileName, ok := profileData.Key.(string)
-		if !ok {
-			return fmt.Errorf("invalid profile name: %v", profileData.Key)
-		}
-		profile := unmarshaledProfiles[profileName]
-		profile.Name = profileName
+	for k, v := range unmarshaledProfiles {
+		profile := v
+		profile.Name = k
```

Comment:
If you're not using the type `type Profiles []*Profile` anymore it can be removed completely. BTW I'm not sure, it may be helpful to automate the unmarshaling (see my other comments)